client/components/indeterminate-checkbox/index.js

```diff
@@ -0,0 +1,27 @@
+import React, { PropTypes } from 'react';
+import Gridicon from 'components/gridicon';
+
+const IndeterminateCheckbox = ( { selectedCount, allCount, onToggle } ) => {
+	const allSelected = selectedCount === allCount;
+	const someSelected = 0 < selectedCount && allCount > selectedCount;
+
+	const onClick = ( event ) => {
+		const checked = ! ( someSelected || allSelected );
```

Comment:
I feel like this logic should be in the reducer, instead of passing down one extra param.